src/node_api.cc

```diff
@@ -920,13 +920,13 @@ NAPI_NO_RETURN void napi_fatal_error(const char* location,
                                      size_t message_len) {
   char* location_string = const_cast<char*>(location);
   char* message_string = const_cast<char*>(message);
-  if (location_len != -1) {
+  if (static_cast<int>(location_len) != -1) {
     location_string = reinterpret_cast<char*>(
         malloc(location_len * sizeof(char) + 1));
```

Comment:
Aside: `sizeof(char)` is 1 by definition and `reinterpret_cast` is overkill here, `static_cast` would have sufficed.

What's more, I don't understand why this code makes copies.  This function does not return.  It seems like completely unneeded complexity.

Reply:
The reason for doing copy here is to account for strings that are not null terminated.

Reply:
Okay, but the other two points still stand.  As well, you don't handle nullptr return values.  If you use `node::Malloc()`, that's taken care of.

Or use `std::string`, that will also stop Coverity from complaining about memory leaks.